src/autocomplete/databases/yql/yql-autocomplete.ts

```diff
@@ -0,0 +1,439 @@
+                context.an_id()?.getText() ?? context.an_id_as_compat()?.getText(),
+            );
+        } catch (error) {
+            if (!(error instanceof c3.DuplicateSymbolError)) {
+                throw error;
+            }
+        }
+
+        return this.visitChildren(context) as {};
+    };
+    visitObject_ref = (context: Object_refContext): {} => {
+        try {
+            this.symbolTable.addNewSymbolOfType(TableSymbol, this.scope, context.getText());
+        } catch (error) {
+            if (!(error instanceof c3.DuplicateSymbolError)) {
+                throw error;
+            }
+        }
+
+        return this.visitChildren(context) as {};
+    };
+    visitNamed_column = (context: Named_columnContext): {} => {
+        try {
+            const alias = context.an_id()?.getText();
+
+            if (alias) {
+                this.symbolTable.addNewSymbolOfType(ColumnAliasSymbol, this.scope, alias);
+            }
+        } catch (error) {
+            if (!(error instanceof c3.DuplicateSymbolError)) {
+                throw error;
+            }
+        }
+
+        return this.visitChildren(context) as {};
+    };
+    visitNamed_expr = (context: Named_exprContext): {} => {
+        try {
+            const alias = context.an_id_or_type()?.getText();
+
+            if (alias) {
+                this.symbolTable.addNewSymbolOfType(ColumnAliasSymbol, this.scope, alias);
+            }
+        } catch (error) {
+            if (!(error instanceof c3.DuplicateSymbolError)) {
+                throw error;
+            }
+        }
+
+        return this.visitChildren(context) as {};
+    };
+    visitResult_column = (context: Result_columnContext): {} => {
+        try {
+            const alias =
+                context.an_id_or_type()?.getText() ?? context.an_id_as_compat()?.getText();
+
+            if (alias) {
+                this.symbolTable.addNewSymbolOfType(ColumnAliasSymbol, this.scope, alias);
+            }
+        } catch (error) {
+            if (!(error instanceof c3.DuplicateSymbolError)) {
+                throw error;
+            }
+        }
+
+        return this.visitChildren(context) as {};
+    };
+}
+
+function processVisitedRules(
+    rules: c3.CandidatesCollection['rules'],
+    cursorTokenIndex: number,
+    tokenStream: TokenStream,
+): ProcessVisitedRulesResult<YQLAutocompleteResult> {
+    let suggestEntity: YQLAutocompleteResult['suggestEntity'];
+    let suggestUdfs = false;
+    let suggestAggregateFunctions = false;
+    let suggestFunctions = false;
+    let suggestWindowFunctions = false;
+    let suggestTableFunctions = false;
+    let suggestPragmas = false;
+    let shouldSuggestColumns = false;
+    let suggestSimpleTypes = false;
+    let shouldSuggestColumnAliases = false;
+
+    const isCreateEntity = getPreviousToken(
+        tokenStream,
+        tokenDictionary,
+        cursorTokenIndex,
+        YQLParser.CREATE,
+    );
```

Comment:
It looks alright, but just so you know, this function is unstable. It's much better to modify grammar instead of looking in the token stream, because token stream doesn't have any ast context

Reply:
What do you think it we release a new version with this implementation so services may start using it and bring me a bugs, and I will modify grammar to make it the best way?